clients/ui/bff/Makefile

```diff
@@ -54,8 +57,18 @@ LOCALBIN ?= $(shell pwd)/bin
 $(LOCALBIN):
 	mkdir -p $(LOCALBIN)
 
+## Tool Binaries
+ENVTEST ?= $(LOCALBIN)/setup-envtest-$(ENVTEST_VERSION)
 GOLANGCI_LINT ?= $(LOCALBIN)/golangci-lint-$(GOLANGCI_LINT_VERSION)
+
+## Tool Versions
 GOLANGCI_LINT_VERSION ?= v1.57.2
+ENVTEST_VERSION ?= release-0.17
+
+.PHONY: envtest
+envtest: $(ENVTEST) ## Download setup-envtest locally if necessary.
+$(ENVTEST): $(LOCALBIN)
+	$(call go-install-tool,$(ENVTEST),sigs.k8s.io/controller-runtime/tools/setup-envtest,$(ENVTEST_VERSION))
```

Comment:
Should the run command try to do this if the mock-k8s-client option is passed to the makefile?

If you try to run `make run MOCK_K8S_CLIENT=true MOCK_MR_CLIENT=true` before the binary is installed you get:

```
time=2024-10-21T15:23:49.306-04:00 level=ERROR msg="failed to start test environment" error="unable to start control plane itself: failed to start the controlplane. retried 5 times: fork/exec /home/gsulliva/Programming/model-registry/clients/ui/bff/bin/k8s/1.29.0-linux-amd64/etcd: no such file or directory"
```

Reply:
Nice catch! Fixed!